RoutesApp_ios/repository/local/syncData/syncData.swift

```diff
@@ -0,0 +1,130 @@
+                    let delete = self.localDataManager.deleteEntityObjectByKeyValue(type: LineEntity.self, key: "id", value: line.id)
+                    if  delete == true {
+                        let newLine = Lines(categoryRef: line.categoryRef, enable: line.enable, id: line.id, idCity: line.idCity,
+                                            idCategory: line.idCategory, name: line.name, updateAt: line.updateAt, createAt: line.createAt)
+                        let lineEntity = newLine.toEntity(categories: self.categoryEntities, context: self.context)
+                        self.updateRoutesByDate(line: lineEntity)
+                    }
+                }
+            case.failure:
+                self.update = false
+            }
+        }
+    }
+    func updateRoutesByDate(line: LineEntity) {
+        lineRouteManager.getLineRouteByDate(idLine: line.id, date: self.lastUpdated) { result in
+            switch result {
+            case.success(let lineRoutes):
+                for route in lineRoutes {
+                    let delete = self.localDataManager.deleteEntityObjectByKeyValue(type: LineRouteEntity.self, key: "id", value: route.id)
+                    if  delete == true {
+                        route.toEntity(context: self.coreDataManager.getContext(), line: line)
+                    }
+                }
+                self.update = true
+                try? self.context.save()
+                self.updateDataValueForSync()
+            case.failure:
+                self.update = false
+            }
+        }
+    }
+    func updateTourPoint() {
+        if coreDataManager.deleteTourPoints() {
+            tourpointsManager.getTourpointList { result in
+                switch result {
+                case.success(let tourpoints):
+                    self.update = true
+                    self.updateTourpointCategories(tourpoints: tourpoints)
+                case.failure:
+                    self.update = false
+                }
+            }
+        }
+    }
+     func updateTourpointCategories(tourpoints: [Tourpoint]) {
+         tourpointsManager.getTourpointCategoriesByDate(date: self.lastUpdated) { result in
+             switch result {
+             case.success(let categories):
+                 let categoryEntities = categories.map({ $0.toEntity(context: self.coreDataManager.getContext()) })
+                 for tourpoint in tourpoints {
+                     tourpoint.toEntity(context: self.coreDataManager.getContext(), categories: categoryEntities)
+                 }
+                 self.update = true
+                 try? self.context.save()
+                 self.updateDataValueForSync()
+             case.failure:
+                 self.update = false
+             }
+         }
+    }
+    func updateDataValueForSync() {
+        if update {
+            self.localDataManager.updateDataValueForSync(entity: "SyncUnitHistory", key: "id", keyValue: "0", keyUpdate: "lastUpdated") { _ in }
```

Comment:
use the constant you already have created

Reply:
done